### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Camelpilot33**
### Describe your change:

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change? -> Grammar and spelling change on CONTRIBUTING.md

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
 - This part is not applicable:
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by imSanko — APPROVED** [review]
Improved grammers ??

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ GitHub will use this tag to [auto-close the issue](https://docs.github.com/en/is
 
 An Algorithm is one or more functions (or classes) that:
 * take one or more inputs,
-* perform some internal calculations or data manipulations,
+* Perform some internal calculations or data manipulations,
```
Comment (jazelly) [inline CONTRIBUTING.md:45]: I think this should be lower cased as it's following a comma

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## Before contributing
 
-Welcome to [TheAlgorithms/Python](https://github.com/TheAlgorithms/Python)! Before sending your pull requests, make sure that you __read the whole guidelines__. If you have any doubt on the contributing guide, please feel free to [state it clearly in an issue](https://github.com/TheAlgorithms/Python/issues/new) or ask the community in [Gitter](https://gitter.im/TheAlgorithms/community).
+Welcome to [TheAlgorithms/Python](https://github.com/TheAlgorithms/Python)! Before sending your pull requests, make sure that you __read the whole guidelines__. If you have any doubt about the contributing guide, please feel free to [state it clearly in an issue](https://github.com/TheAlgorithms/Python/issues/new) or ask the community in [Gitter](https://gitter.im/TheAlgorithms/community).
```
Comment (jazelly) [inline CONTRIBUTING.md:5]: "on" sounds correct to me

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -168,15 +168,15 @@ We want your work to be readable by others; therefore, we encourage you to note
 
   Instructions on how to install mypy can be found [here](https://github.com/python/mypy). Please use the command `mypy --ignore-missing-imports .` to test all files or `mypy --ignore-missing-imports path/to/file.py` to test a specific file.
 
-- [__List comprehensions and generators__](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions) are preferred over the use of `lambda`, `map`, `filter`, `reduce` but the important thing is to demonstrate the power of Python in code that is easy to read and maintain.
+- [__List comprehensions and generators__](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions) are preferred over the use of `lambda`, `map`, `filter`, and `reduce` but the important thing is to demonstrate the power of Python in code that is easy to read and maintain.
```
Comment (tianyizheng02) [inline CONTRIBUTING.md:171]:
```suggestion
- [__List comprehensions and generators__](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions) are preferred over the use of `lambda`, `map`, `filter`, `reduce`, etc. but the important thing is to demonstrate the power of Python in code that is easy to read and maintain.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -168,15 +168,15 @@ We want your work to be readable by others; therefore, we encourage you to note
...
 
 - Avoid importing external libraries for basic algorithms. Only use those libraries for complicated algorithms.
 - If you need a third-party module that is not in the file __requirements.txt__, please add it to that file as part of your submission.
 
 #### Other Requirements for Submissions
 - If you are submitting code in the `project_euler/` directory, please also read [the dedicated Guideline](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md) before contributing to our Project Euler library.
 - The file extension for code files should be `.py`. Jupyter Notebooks should be submitted to [TheAlgorithms/Jupyter](https://github.com/TheAlgorithms/Jupyter).
-- Strictly use snake_case (underscore_separated) in your file_name, as it will be easy to parse in future using scripts.
+- Strictly use snake_case (underscore_separated) in your file_name, as it will be easy to parse in the future using scripts.
```
Comment (tianyizheng02) [inline CONTRIBUTING.md:179]: British English makes a distinction between "in the future" and "in future"

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
Most of these changes correct "mistakes" that aren't actually incorrect, so please change them

**Comment by tianyizheng02** [comment]
Closing this PR because
1. OP hasn't responded in over 6 months, so the PR may as well be stale
2. Almost all of the "fixes" were unnecessary, as most changes didn't fix any real mistakes

**Pull request closed without merging**